snuba/migrations/parse_schema.py

```diff
@@ -0,0 +1,152 @@
+import logging
+import re
+
+from parsimonious.grammar import Grammar   # type: ignore
+from parsimonious.nodes import Node, NodeVisitor  # type: ignore
+from typing import Any, Iterable, Mapping, Sequence, Tuple
+
+from snuba.clickhouse.columns import (
+    Array,
+    ColumnType,
+    Date,
+    DateTime,
+    Enum,
+    FixedString,
+    Float,
+    IPv4,
+    IPv6,
+    LowCardinality,
+    Materialized,
+    Nullable,
+    String,
+    UInt,
+    UUID,
+    WithCodecs,
+    WithDefault,
+)
+
+logger = logging.getLogger("snuba.migrate")
+
+
+grammar = Grammar(
+    r"""
+    type             = primitive / lowcardinality / nullable / array
+    primitive        = basic_type / uint / float / fixedstring / enum
+    # DateTime must come before Date
+    basic_type       = "DateTime" / "Date" / "IPv4" / "IPv6" / "String" / "UUID"
+    uint             = "UInt" uint_size
+    uint_size        = "8" / "16" / "32" / "64"
+    float            = "Float" float_size
+    float_size       = "32" / "64"
+    fixedstring      = "FixedString" open_paren fixedstring_size close_paren
```

Comment:
You should always leave an optional sequence of spaces after ( and before ) and  before and after commas. Clickhosue is probably not going to format its expressions by adding spaces but they would be valid expressions

Reply:
Makes sense, done